clients/src/test/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumerTest.java

```diff
@@ -604,8 +605,6 @@ public void testCommitSyncLeaderEpochUpdate() {
 
         assertDoesNotThrow(() -> consumer.commitSync(topicPartitionOffsets));
 
-        verify(metadata).updateLastSeenEpochIfNewer(t0, 2);
-        verify(metadata).updateLastSeenEpochIfNewer(t1, 1);
```

Comment:
Out of curiosity, why don't we want to verify the metadata was updated? That still happens in the `ApplicationEventProcessor`, right?

Reply:
We mock `ApplicationEventProcessor` in `AsyncKafkaConsumerTest`, so can't verify related implementation in it. We test it in https://github.com/apache/kafka/pull/17150/files#diff-87404aabc3e6c99569d678ff904586ab232577240d9932839c52135a34b2b57aR260.